librepo/downloader.c

```diff
@@ -684,7 +684,7 @@ lr_writecb(char *ptr, size_t size, size_t nmemb, void *userdata)
     }
 
     assert(nmemb > 0);
-    cur_written = fwrite(ptr, size, nmemb, target->f);
+    cur_written = write(target->fd, ptr, size * nmemb);
```

Comment:
What if size*nmemb overflows size_t? fwrite() handled it for you. Now you need to implement it check yourself.

Reply:
Updated to check for overflow and fall back to a well known size.